appsmithorg/appsmith
# [Task]: Update DataTreeFactory to include widget resizes and remove references to layout mode and viewport.

## Description

- Remove ```appPositioningType``` and ```isMobile``` props from widget DataTree.
- Include width and height as ```derivedHeight and derivedWidth``` in widget DataTree. Prefer width and height if available, else calculate using position props.
- Update TableWidgetV2 to use the ```derivedHeight``` prop in derivedProps calculation to get pageSize.

#### PR fixes following issue(s)
Fixes # [26629](https://github.com/appsmithorg/appsmith/issues/26629)


#### Type of change
> Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

## Testing
>
#### How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Also list any relevant details for your test configuration.
> Delete anything that is not relevant
- [ ] JUnit

#### Test Plan
> Add Testsmith test cases links that relate to this PR
1. Create an app.
2. Add a Table widget.
3. Connect the Table widget to a datasource.
4. Assert that the Table widget works as expected. The number of rows per page are calculated accurately based on initial height of the widget.

## Checklist:
#### Dev activity


#### QA activity:
